CHANGELOG.md

```diff
@@ -1,5 +1,16 @@
 # Changelog
 
+## [v2.0.0-ALPHA-1](https://github.com/stargate/stargate/tree/v2.0.0-ALPHA-1) (2022-01-24)
+
+[Full Changelog](https://github.com/stargate/stargate/compare/v1.0.47...v2.0.0-ALPHA-1)
+
+**Merged pull requests:**
+
+- add separate release workflow for v2 [\#1574](https://github.com/stargate/stargate/pull/1574) ([jeffreyscarpenter](https://github.com/jeffreyscarpenter))
+- Bumping version for next release [\#1570](https://github.com/stargate/stargate/pull/1570) ([github-actions[bot]](https://github.com/apps/github-actions))
+- restart docker before running tests on CI [\#1567](https://github.com/stargate/stargate/pull/1567) ([ivansenic](https://github.com/ivansenic))
+- refactor update document sub-path to use new write service [\#1562](https://github.com/stargate/stargate/pull/1562) ([ivansenic](https://github.com/ivansenic))
+
```

Comment:
still not getting how would this work once v2 is rebased on the master, how would you merge the change log?

Reply:
I think of this in terms of the desired end state - how the changelog should be structured when v2 is the official version. It should look something like this (numbers are just for illustration purposes):

- v2.0.0 
- v2.0.0-ALPHA-3
- v2.0.0-ALPHA-2
- v2.0.0-ALPHA-1
- v1.0.50
- v1.0.49
- ...

I think this is a way of rendering things that will make the most sense to someone who is trying to read the changelog. Even though some of the v1 and v2 releases are interleaved in terms of the time they are generated, we will always be merging the latest 1.x release when we generate a 2.x release, so any v2 release will cumulatively include all prior v1 releases listed. 

The changelog generation tool that we are using does not correctly handle this type of situation. I have attempted some workarounds to attempt to get it to generate correctly but because of its time-based algorithm it does not work correctly. I will generate the v2.0.0-ALPHA-1 PR list manually and submit an updated changelog to this PR. 

I am hoping that once we have an initial v2 changelog, the tool will function correctly for subsequent v2 releases, but we will have to see how it goes. The changelog generation process will likely be cleaner if we start the practice of doing a v2 release corresponding to each v1 release. This should be possible since we have agreed to start requiring green IT runs for merging to the `v2.0.0` branch.

From the output that you saw in this commit, it appears that PRs that are merged to from v1 to v2 are duplicated in the v2 portion of the changelog. I think it will be best to delete those duplicates, either manually or perhaps we will write a script to de-duplicate.  

In terms of your original question, which I think is how the changelog will be generated when v2 is the official version, this will require more investigation and discussion. I think we should seriously consider not merging the `v2.0.0` branch back to `master` but instead renaming the `v2.0.0` branch to `main` and making it the official version, as I mentioned in comment on [this discussion](#1489).

Reply:
I updated the changelog to include a manually generated section for v2.0.0-ALPHA-1, which I created using the following:

`git cherry -v master | nl | sort -nr | awk '{print "- " substr($0,index($0,$4))}' | grep "(#"`

This pulls commits that are unique to the `v2.0.0` branch, inverts the sort order to be most recent first, does some simple line formatting, and eliminates commits not tied to a PR.